src/compiler/rust/canister_methods/src/hash_file.rs

```diff
@@ -0,0 +1,107 @@
+use quote::quote;
+
+pub fn get_hash_file() -> proc_macro2::TokenStream {
+    quote! {
+        pub fn hash_file(path: String) {
+            ic_cdk::println!("START HASH FILE");
+            hash_file_by_parts(path, 0)
+        }
+
+        #[ic_cdk_macros::query]
+        pub fn get_file_hash(path: String) -> String {
+            ASSETS_HASHES
+                .with(|asset_hashes| asset_hashes.borrow().get(&path).unwrap().clone())
+                .iter()
+                .map(|bytes| format!("{:02x}", bytes))
+                .collect()
+        }
+
+        #[ic_cdk_macros::query]
+        pub fn get_file_hashes() -> Vec<String> {
+            ASSETS_HASHES
+                .with(|asset_hashes| asset_hashes.borrow().clone())
+                .iter()
+                .map(|(path, bytes)| {
+                    format!(
+                        "{}: {}",
+                        path,
+                        bytes
+                            .iter()
+                            .map(|bytes| format!("{:02x}", bytes))
+                            .collect::<String>()
+                    )
+                })
+                .collect()
+        }
+
+        fn hash_file_by_parts(path: String, position: u64) {
+            ic_cdk::println!(
+                "Hash {} starting at: {}",
+                path,
+                bytes_to_human_readable(position)
+            );
+            let mut file = std::fs::File::open(&path).unwrap();
+
+            std::io::Seek::seek(&mut file, std::io::SeekFrom::Start(position)).unwrap();
```

Comment:
I wonder if there's a better way to do this

Reply:
I can't find one. Mostly this looks bad because I wanted to qualify the methods instead of importing them